docs/source/jupyter/hpc-sdk.ipynb

```diff
@@ -0,0 +1,432 @@
+    "The name can be either `iwm` or `mc` for two deployments. \n",
+    "\n",
+    "Note the `iwm` deployment **don't** have the slurm running properly at the moment, therefore the job submit is not working, but all other capabilities are working.\n",
+    "\n",
+    "The EPFL Materials Cloud (mc) deployment support all capabilities and app developers can use it embeded in the app that need to running the heavy calculations. \n",
+    "\n",
+    "To initialize the instance, provide the deployment name and MarketPlace `access_token`. \n",
+    "The `access_token` can be inherit from the App which integrate the hpc gateway app as the calculation backend.\n",
+    "\n",
+    "To run this notebook, put the `.env` file with `ACCESS_TOKEN` set in the same folder."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from hpc_gateway_sdk import get_app\n",
+    "from dotenv import load_dotenv\n",
+    "import os\n",
+    "\n",
+    "load_dotenv(\".env\")\n",
+    "\n",
+    "access_token = os.environ.get(\"ACCESS_TOKEN\")\n",
+    "app = get_app(name=\"mc\", access_token=access_token)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The first time using the HPC gateway app, you need to create the user in the database of HPC app to record the job data conrespond to every MarketPlace user account.\n",
+    "Meanwhile, `create_user` will create the user folder in the cluster to store jobs folder repository."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{\n",
+      "  \"_id\": \"638f355e57bd4aa2a97b98d0\", \n",
+      "  \"email\": \"jusong.yu@epfl.ch\", \n",
+      "  \"home\": \"/scratch/snx3000/jyu/firecrest/jusong_yu\", \n",
+      "  \"message\": \"Success: Create user in database.\", \n",
+      "  \"name\": \"Jusong Yu\"\n",
+      "}\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "user_info = app.create_user()\n",
+    "print(user_info)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To create a job, using `create_job` endpoint of the gateway app.\n",
+    "It will create a job folder in the remote cluster to store files and to submit job.\n",
+    "The `jobid` is returned for further operations.\n",
+    "The parameter `new_transformation` is a dict for the job information and is used to create the slurm job script.\n",
+    "The following parameters must be provide to create the job.\n",
+    "\n",
+    "- `job_name`: the name of the job.\n",
+    "- `ntasks_per_node`: the numper of tasks per node a.k.a the mpi stacks of your job, is the number follow the `mpirun -n`.\n",
+    "- `partition`: for the EPFL Materials Cloud (mc) deployment, the partition can be used are `debug` and `normal`. \n",
```

Comment:
But what does `partition` really mean? Will they also be available for the IWM cluster?

Reply:
If the IWM cluster doesn't have this, we can not include it. But this is a very typical setting for slurm to distinguish the different resources.

Reply:
That was my question, does the IWM cluster have it?
Also, users are probably not familiar with Slurm at all, so we can't assume they will know what it is.

Reply:
> does the IWM cluster have it

When I tested, it did not have this. You are correct, this parameter better be removed and set with a default value. It is not very flexible for the payload passed to `create_job`.

Can we keep it for now, since it require to change both here in the doc and the implementation of HPC app.

Reply:
I would suggest making an issue on the repo with the implementation (same as with the folder name) so that we can work on them once the branch situation is a bit clearer :)

Reply:
Issues https://github.com/materials-marketplace/hpc-gateway-app/issues/11 and https://github.com/materials-marketplace/hpc-gateway-app/issues/12 creaded.